src/components/custom-field-input-text/custom-field-input-text.jsx

```diff
@@ -38,6 +40,27 @@ export default function CustomFieldInputText(props) {
     }
   });
 
+  const icon = () => {
+    if (props.error) {
+      return (<Icon
+        className={styles['input-icon']}
+        currentColor="caution"
+        name={cautionSvg.id}
+        size="medium"
+      />);
+    }
+
+    if (props.icon) {
+      return props.icon;
+    }
+
+    return undefined;
+  };
+
+  const showIcon = () => {
+    return props.error || !!props.icon;
```

Comment:
I'm not sure we need to coerce `props.icon` here, considering our default is `undefined`, and type is `PropTypes.node`.

Reply:
We show an error icon if we have an error, or if we have an icon handed to us we present it. This is how we show the calendar icon